tmt/base.py

```diff
@@ -1466,24 +1494,121 @@ def show(self):
         # Prepare absolute workdir path if --id was used
         id_ = self.opt('id')
         path = self.opt('path')
-        if id_ and '/' not in id_:
-            id_ = os.path.join(path, id_)
         self.print_header()
-        for filename in os.listdir(path):
-            abs_path = os.path.join(path, filename)
-            invalid_id = id_ and abs_path != id_
-            invalid_run = not os.path.exists(
-                os.path.join(abs_path, 'run.yaml'))
-            if not os.path.isdir(abs_path) or invalid_id or invalid_run:
-                continue
-            # Creating a and loading a run may override the data in the
-            # context which could affect the status of the following runs.
-            # Backup the inner context object to later recover it to
-            # its initial state.
-            backup = copy.deepcopy(self._context.obj)
-            run = Run(abs_path, self._context.obj.tree, self._context)
+        for abs_path in tmt.utils.generate_runs(path, id_):
+            run = Run(abs_path, self._context.obj.tree, self._context, False)
             self.process_run(run)
-            self._context.obj = backup
+
+
+class Clean(tmt.utils.Common):
+    """ A class for cleaning up workdirs, guests or images """
```

Comment:
Would be nice to add the new class to the [overview](https://tmt.readthedocs.io/en/latest/contribute.html#class-structure).